src/util/types.ts

```diff
@@ -0,0 +1 @@
+export type ArrayElement<T> = T extends ArrayLike<infer U> ? U : T;
```

Comment:
Is this used anywhere?

Reply:
I think it is. I was trying to bring even more type safety to infer element type from List and Vector. Will check